pandas/core/indexes/period.py

```diff
@@ -534,6 +534,22 @@ def shift(self, periods: int = 1, freq=None) -> Self:
             )
         return self + periods
 
+    def _convert_can_do_setop(self, other):
+        if (
+            not isinstance(other, Index)
+            or other.empty
+            or isinstance(other, PeriodIndex)
+        ):
+            return super()._convert_can_do_setop(other)
+
+        # Convert non-PeriodIndex to PeriodIndex
+        try:
+            other = PeriodIndex(other, freq=self.freq)
```

Comment:
i dont think we should be doing this casting implicitly.  the user should do it explicitly if they want it.